buildbot_nix/__init__.py

```diff
@@ -44,6 +44,57 @@ class BuildbotNixError(Exception):
     pass
 
 
+class BuildbotEffectsTrigger(Trigger):
+    """Dynamic trigger that run buildbot effect defined in a flake."""
+
+    def __init__(
+        self,
+        effects_scheduler: str,
+        effects: list[dict[str, Any]],
+        **kwargs: Any,
+    ) -> None:
+        if "name" not in kwargs:
+            kwargs["name"] = "trigger"
+        self.effects = effects
+        self.config = None
+        self.effects_scheduler = effects_scheduler
+        super().__init__(
+            self,
+            waitForFinish=True,
+            schedulerNames=[effects_scheduler],
+            haltOnFailure=True,
+            flunkOnFailure=True,
+            sourceStamps=[],
+            alwaysUseLatest=False,
+            updateSourceStamp=False,
+            **kwargs,
+        )
+
+    def createTriggerProperties(self, props: Any) -> Any:  # noqa: N802
+        return props
+
+    def getSchedulersAndProperties(self) -> list[tuple[str, Properties]]:  # noqa: N802
+        build_props = self.build.getProperties()
+        repo_name = build_props.getProperty(
+            "github.base.repo.full_name",
+            build_props.getProperty("github.repository.full_name"),
+        )
+        project_id = slugify_project_name(repo_name)
+        source = f"buildbot-run-effect-{project_id}"
+
+        triggered_schedulers = []
+        for effect in self.effects:
+            props = Properties()
+            props.setProperty("virtual_builder_name", effect, source)
+            props.setProperty("status_name", f"hci-effects.{effect}", source)
```

Comment:
```suggestion
            props.setProperty("status_name", f"effects.{effect}", source)
```

Is `outputs.hci-effects` instead of `outputs.effects` intentional?

Reply:
I think hercules-ci-effects did rename this.

Reply:
https://github.com/hercules-ci/hercules-ci-effects/blob/31b6cd7569191bfcd0a548575b0e2ef953ed7d09/flake-public-outputs.nix#L24

Reply:
Not quite sure which one is right: https://docs.hercules-ci.com/hercules-ci-effects/guide/import-or-pin.html

Reply:
Ah. There seem to be different outputs: https://flake.parts/options/hercules-ci-effects.html#opt-herculesCI

Reply:
The default hercules schema is `outputs.effects`: https://github.com/hercules-ci/hercules-ci-agent/blob/429ef2392873f2d249096bad07250ce73c88a736/hercules-ci-agent/data/default-herculesCI-for-flake.nix#L180

Some of the preconfigured effects use `outputs.effects`, e.g https://github.com/hercules-ci/hercules-ci-effects/blob/31b6cd7569191bfcd0a548575b0e2ef953ed7d09/flake-modules/github-pages.nix#L127

Reply:
Sorry, when I said `outputs.effects` I meant `herculesCI.onPush.default.outputs.effects`.

Reply:
```diff
diff --git a/buildbot_effects/__init__.py b/buildbot_effects/__init__.py
index 30ddf28..abd4e13 100644
--- a/buildbot_effects/__init__.py
+++ b/buildbot_effects/__init__.py
@@ -96,7 +96,7 @@ def effect_function(opts: EffectsOptions) -> str:
     rev = args["rev"]
     escaped_args = json.dumps(json.dumps(args))
     url = json.dumps(f"git+file://{opts.path}?rev={rev}#")
-    return f"""(((builtins.getFlake {url}).outputs.effects (builtins.fromJSON {escaped_args})))"""
+    return f"""(((builtins.getFlake {url}).outputs.herculesCI (builtins.fromJSON {escaped_args})).onPush.default.outputs.effects)"""
 
 
 def list_effects(opts: EffectsOptions) -> list[str]:
diff --git a/buildbot_nix/__init__.py b/buildbot_nix/__init__.py
index 3633b42..fee8023 100644
--- a/buildbot_nix/__init__.py
+++ b/buildbot_nix/__init__.py
@@ -86,7 +86,7 @@ class BuildbotEffectsTrigger(Trigger):
         for effect in self.effects:
             props = Properties()
             props.setProperty("virtual_builder_name", effect, source)
-            props.setProperty("status_name", f"hci-effects.{effect}", source)
+            props.setProperty("status_name", f"effects.{effect}", source)
             props.setProperty("virtual_builder_tags", "", source)
             props.setProperty("command", effect, source)
 

```